hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/S3EventsHoodieIncrSource.java

```diff
@@ -217,11 +220,18 @@ public Pair<Option<Dataset<Row>>, String> fetchNextBatch(Option<String> lastCkpt
 
     Option<Dataset<Row>> dataset = Option.empty();
     if (!cloudFiles.isEmpty()) {
-      DataFrameReader dataFrameReader = getDataFrameReader(fileFormat);
-      dataset = Option.of(dataFrameReader.load(cloudFiles.toArray(new String[0])));
+      JavaRDD<Dataset<Row>> datasetIterator = sparkContext.parallelize(cloudFiles, cloudFiles.size()).flatMap(
```

Comment:
Why do we need to take the list of paths and make it into an rdd instead of keeping the list of paths in its rdd form? Instead of calling `collectAsList()` on line 219, could we just leave that as a Dataset<String>?

Reply:
yes, have fixed it.